### PR TITLE
insert: add NEON support

### LINKED ISSUE #691: Implement insert for ARM architecture

### PR DESCRIPTION
**Pull request by amyspark**
👋 

Fixes #691

### CODE REVIEW

File: include/xsimd/arch/xsimd_neon.hpp [hunk 1]
```diff
@@ -1380,6 +1380,40 @@ namespace xsimd
...
+            case 1:
+                return vsetq_lane_u8(val, self, I);
+            case 2:
+                return vsetq_lane_u16(val, self, I);
+            case 4:
+                return vsetq_lane_u32(val, self, I);
+            case 8:
+                return vsetq_lane_u64(val, self, I);
+            default:
+                return insert(val, self, pos, neon {});
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_neon.hpp:1401]: I don't understand that default o_O

Comment (amyspark) [reply include/xsimd/arch/xsimd_neon.hpp]: Just fixed the typo! Excess of copy-paste there.

File: include/xsimd/arch/xsimd_neon.hpp [hunk 2]
```diff
@@ -1380,6 +1380,40 @@ namespace xsimd
...
+            case 1:
+                return vsetq_lane_u8(val, self, I);
+            case 2:
+                return vsetq_lane_u16(val, self, I);
+            case 4:
+                return vsetq_lane_u32(val, self, I);
+            case 8:
+                return vsetq_lane_u64(val, self, I);
+            default:
+                return insert(self, val, pos, neon {});
```
Comment (serge-sans-paille) [inline include/xsimd/arch/xsimd_neon.hpp:1401]: That's only introducing a recursion. You're convering all supported size, so an `assert(false && "unsupported size")` should be enough here

Comment (amyspark) [reply include/xsimd/arch/xsimd_neon.hpp]: I meant the catch-all generic fallback, I've just added it.

**Pull request merged**